test/integration_test/basic_test.go

```diff
@@ -243,3 +270,79 @@ func testInstallWithTelemetry(t *testing.T, cluster *corev1.StorageCluster) {
 	err = coreops.Instance().DeleteSecret(secret.Name, secret.Namespace)
 	require.NoError(t, err)
 }
+
+// BasicCsiRegression test includes the following steps:
+// 1. Deploy PX with CSI enabled and validate CSI components and images
+// 2. Delete "px-csi-ext" pods and validate they get re-deployed
+// 3. Disable CSI and validate CSI components got successfully removed
+// 4. Enabled CSI and validate CSI components and images
+// 5. Delete "portworx" pods and validate they get re-deployed
+// 6. Delete "px-csi-ext" pods and validate they get re-deployed
+// 7. Delete StorageCluster and validate it got successfully removed
+func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		if tc.ShouldSkip() {
+			t.Skip()
+		}
+
+		var err error
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Delete portworx oci-mon pods and validate they get re-deployed
+		logrus.Info("Delete portworx pods and validate they get re-deployed")
+		//err = testutil.DeletePodsByLabels(cluster.Namespace, map[string]string{"name": "portworx"})
+		err = coreops.Instance().DeletePodsByLabels(cluster.Namespace, map[string]string{"name": "portworx"}, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
+		require.NoError(t, err)
+
+		// Delete px-csi-ext pods and validate they get re-deployed
+		logrus.Info("Delete px-csi-ext pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("px-csi-ext", cluster.Namespace, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
+		require.NoError(t, err)
+
+		// Disable CSI and update live StorageCluster
+		logrus.Info("Disable CSI and validate StorageCluster")
+		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+		require.NoError(t, err)
+		cluster.Spec.FeatureGates = map[string]string{"CSI": "false"}
+		cluster, err = ci_utils.UpdateStorageCluster(cluster)
+		require.NoError(t, err)
+
+		// Sleep for 20 seconds to let operator start the update process
+		logrus.Debug("Sleeping for 20 seconds...")
+		time.Sleep(20 * time.Second)
+
+		// Validate cluster deployment
+		logrus.Infof("Validate StorageCluster %s", cluster.Name)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateUpgradeTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
```

Comment:
I guess you wanted to use `ci_utils.DefaultValidateDeployTimeout` and not `ci_utils.DefaultValidateUpgradeTimeout`.

Reply:
Done, made a new timeout vars for Update